test/test_sbol2_sbol3_direct.py

```diff
@@ -27,18 +27,25 @@ def handle_2to3_conversion(self, test_filename, rubric_filename):
 
         # Convert to SBOL3 and check contents
         doc3 = convert2to3(doc2, use_native_converter=True)
-        self.assertEqual(len(doc3.validate()), 0)
+        report = doc3.validate()
+        self.assertEqual(len(report), 0, f'Validation failed: {report}')
         with tempfile.TemporaryDirectory() as tmpdir:
             tmp3 = Path(tmpdir) / 'doc3.nt'
             doc3.write(tmp3)
             self.assertFalse(file_diff(str(tmp3), str(TEST_FILES / rubric_filename)))
+
+            # Round-trip back to SBOL2 and check contents
             doc2_loop = convert3to2(doc3, True)
+            report = doc2_loop.validate()
+            self.assertEqual(report, 'Valid.', f'Validation failed: {report}')
             tmp2 = Path(tmpdir) / 'doc2_loop.xml'
             doc2_loop.write(tmp2)
             self.assertFalse(file_diff(str(tmp2), str(TEST_FILES / test_filename)))
-            report = doc3.validate()
-            self.assertEqual(len(report), 0, f'Validation failed: {report}')
 
+    # Currently fails with the following validation error:
+    # sbol-12403: Strong Validation Error: The endedAtTime property of an Activity object is OPTIONAL and MAY contain
+    # a DateTime. Reference: SBOL Version 2.3.0 Section 12.8.1 on page 73 : https://synbiohub.org/public/igem/igem2sbol
+    @unittest.expectedFailure
```

Comment:
Rather than turning off this test, I think that it would be good to check that precisely this error happens and no others.

Reply:
This datetime value in the current test file (BBa_J23101.xml ) fails validation: 
`    <prov:endedAtTime>2017-03-06T15:00:00+00:00</prov:endedAtTime>`

However, a datetime that includes milliseconds _will_ validate.  (I figured this out by looking at test files in SBOLTestSuite that include a valid `endedAtTime`):
`    <prov:endedAtTime>2017-03-06T15:00:00.000+00:00</prov:endedAtTime>`

So, you may be correct that this is a problem with SBOLValidator, since these datetimes are semantically equivalent.

The most expedient fix is for me to just manually change the datetime in the test file to the accepted format.  Would that be acceptable?

Reply:
I attempted a patching solution (option 1) above but ran into challenges, so I'm working on implementing option 2 instead.  The big obstacle to patching is that the `DateTimePropertyMixin.from_user` `DateTimePropertyMixin.to_user` methods are not exposed in sbol3 (although, in sbol2, the analogous methods _are_ exposed and can be overridden).

I have opened an issue on SBOL-Validator to track this datetime issue:
https://github.com/SynBioDex/SBOL-Validator/issues/142